doc/tutorials/klondike/step2.md

```diff
@@ -0,0 +1,258 @@
+  bool get debugMode => true;
+}
+```
+
+Here we declare the `Stock` class as a `PositionComponent` (which is a component
+that has a position and size). We also turn on the debug mode for this class so
+that we can see it on the screen even though we don't have any rendering logic
+yet.
+
+Likewise, create three more files `components/foundation.dart`,
+`components/pile.dart`, and `components/waste.dart`. For now all four classes
+will have exactly the same logic inside, we'll be adding more functionality into 
+those classes in subsequent chapters.
+
+
+## Game structure
+
+Once we have some basic components, they need to be added to the game. It is 
+time to make a decision about the high-level structure of the game.
+
+There exist multiple approaches here, which differ in their complexity,
+extendability, and overall philosophy. The approach that we will be taking in
+this tutorial is based on using the [World] component, together with a [Camera].
+
+The idea behind this approach is the following: imagine that your game world
+exists independently from the device, that it exists already in our heads, and
+on the sketch, even though we haven't done any coding yet. This world will have
+a certain size, and each element in the world will have certain coordinates. It
+is up to us to decide what will be the size of the world, and what is the unit
+of measurement for that size. The important part is that the world exists
+independently from the device, and its dimensions likewise do not depend on the
+pixel resolution of the screen.
+
+All elements that are part of the world will be added to the `World` component,
+and the `World` component will be then added to the game.
+
+The second part of the overall structure is a camera (`CameraComponent`). The
+purpose of the camera is to be able to look at the world, to make sure that it
+renders at the right size on the screen of the user's device.
+
+Thus, the overall structure of the component tree will look approximately like 
+this:
+```text
+KlondikeGame
+ ├─ World
+ │   ├─ Stock
+ │   ├─ Waste
+ │   ├─ Foundation (×4)
+ │   └─ Pile (×7)
+ └─ CameraComponent
+```
+
+For this game I've been drawing my image assets having in mind the dimension of
+a single card at 1000×1400 pixels. So, this will serve as the reference size for
+determining the overall layout. Another important measurement that affects the
+layout is the inter-card distance. It seems like it should be somewhere between
+150 to 200 units (relative to the card width), so we will declare it as a 
+variable `gap` that can be adjusted later if needed. For simplicity, both the
+vertical and horizontal inter-card distance will be the same, and the minimum
+padding between the cards and the edges of the screen will also be equal to
+`gap`.
+
+Alright, let's put all this together and implement our `KlondikeGame`:
+
+```dart
+class KlondikeGame extends FlameGame {
+   final gap = 175.0;
+   final w = 1000.0;
+   final h = 1400.0;
```

Comment:
Explicit variable types on public variables

Reply:
We also shouldn't encourage short forms of variables, I guess you did it so that it wouldn't clash with `width` and `height` of the game? Or do we only have those on `PositionComponent`? If so, I think you can just rename them to `width` and `height`.

Reply:
I renamed them to `cardWidth`, `cardHeight`.